### ISSUE TITLE
Allow for line-ignore option

### ISSUE THREAD

**Issue by muenzpraeger** [forcedotcom/salesforcedx-slds-lsp#11]
It is a common scenario that CSS classes get constructed within a JavaScript getter.

![image](https://user-images.githubusercontent.com/2205451/80868622-30d0d280-8c9c-11ea-8eec-21e2f6d15ecc.png)

It would be great to have an ignore option like `// slds-ignore-next-line` that allows the developer to disable validation, in the sense like ESLint or Prettier allow to ignore lines.

**Comment by jonnyl** [comment]
Agreed, this will help get rid of false positives especially as we start rolling out the validator in our internal automation.  @kirupa

**Comment by kirupa** [comment]
This is a good suggestion! I will add it to our next sprint’s deliverable 😀

**Comment by doniarobinson-sf** [comment]
Can we think of another valid case for people to ignore lines? Because in the example above, if we let them ignore that line, we don't get a chance to validate that second token. They might also choose to ignore lines instead of fixing them when they receive their reports for what they've done wrong. Maybe I'm cynical, but I can see a lot ways that can go awry, and we actually want a heavy degree of control here. However, another way of resolving the issue listed above is to check for a token that contains a curly brace and not validate that.

**Comment by muenzpraeger** [comment]
The same applies to every other part where the SLDS plugin shows errors, also with a CSS class. The developer should always be in full control of what they fix, and what they want to ignore. That's how every other linting works.

**Comment by ambreeen** [comment]
This is definitely something we can consider, but I do want to add that the SLDS Validator isn't a linter, it is a validator, which is inherently different in their purpose.

**Comment by jonnyl** [comment]
Here is a great example of where we would like to allow this entire file to be ignored by SLDSValidation:
https://gus.lightning.force.com/lightning/r/0D5B000001CBqkIKAT/view

**Comment by eltoroit** [comment]
+1

**Comment by billyma** [comment]
We are working on this item, and will publish the updated behavior soon.

**Comment by SteffenBaumann** [comment]
Any updates on this one?

**Comment by LoganTann-pro** [comment]
Can we also add a setting to ignore files matching a pattern, eg by adding an eslint ignore file support ?

Currently, the forceignore files cannot be used as this might prevent the file we want to ignore to be deployed by sfdx. Changing the eslint ignore file does not affect the lwc validator.
